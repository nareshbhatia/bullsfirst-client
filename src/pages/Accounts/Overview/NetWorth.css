.net-worth__label-value {
  margin-right: 80px;
}

.net-worth__label {
  font-size: 14px;
  line-height: 1;
  text-transform: uppercase;
  color: var(--text-muted);
  margin: 0;
}

.net-worth__value {
  font-size: 24px;
  line-height: 1.1;
  margin: 8px 0 0 0;
}
