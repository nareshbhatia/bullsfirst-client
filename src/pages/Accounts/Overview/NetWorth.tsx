import React, { useEffect } from 'react';
import { gql, useQuery } from '@apollo/client';
import { NumberUtils } from '@react-force/number-utils';
import { useParams } from 'react-router-dom';
import { HorizontalContainer, Loading } from '../../../components';
import { useRefreshContext } from '../../../contexts';
import { GetNetWorth } from './__generated__/GetNetWorth';
import './NetWorth.css';

const { formatAsMoney } = NumberUtils;

export const GET_NET_WORTH = gql`
  query GetNetWorth($accountId: ID!) {
    netWorthInfo(accountId: $accountId) {
      netWorth
      investments
      cash
    }
  }
`;

const LabelValue = ({ label, value }: { label: string; value: number }) => {
  return (
    <div className="net-worth__label-value">
      <h1 className="title2">{label}</h1>
      <p className="net-worth__value">${formatAsMoney(value)}</p>
    </div>
  );
};

export const NetWorth = () => {
  const { accountId } = useParams();
  const { refreshCount } = useRefreshContext();
  const { loading, data, refetch } = useQuery<GetNetWorth>(GET_NET_WORTH, {
    variables: {
      accountId,
    },
  });

  useEffect(() => {
    refetch();
  }, [refreshCount, refetch]);

  if (loading || !data) {
    return <Loading />;
  }

  const { cash, investments, netWorth } = data.netWorthInfo;
  return (
    <HorizontalContainer className="p-2">
      <LabelValue label="Net Worth" value={netWorth} />
      <LabelValue label="Investments" value={investments} />
      <LabelValue label="Cash" value={cash} />
    </HorizontalContainer>
  );
};
