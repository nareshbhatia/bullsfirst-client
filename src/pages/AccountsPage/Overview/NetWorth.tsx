import React, { useEffect } from 'react';
import { useQuery } from '@apollo/client';
import { NumberUtils } from '@react-force/number-utils';
import { useParams } from 'react-router-dom';
import { HorizontalContainer, Loading } from '../../../components';
import { useRefreshContext } from '../../../contexts';
import { GetNetWorthDocument } from '../../../graphql/generated';
import './NetWorth.css';

const { formatAsMoney } = NumberUtils;

const LabelValue = ({ label, value }: { label: string; value: number }) => {
  return (
    <div className="net-worth__label-value">
      <h1 className="title2">{label}</h1>
      <p className="net-worth__value">${formatAsMoney(value)}</p>
    </div>
  );
};

export const NetWorth = () => {
  const { accountId } = useParams();
  const { refreshCount } = useRefreshContext();
  const { loading, error, data, refetch } = useQuery(GetNetWorthDocument, {
    variables: {
      accountId,
    },
  });

  useEffect(() => {
    refetch();
  }, [refreshCount, refetch]);

  if (loading) {
    return <Loading />;
  }
  if (error) {
    throw error;
  }
  if (!data) {
    throw new Error('Something went wrong');
  }

  const { cash, investments, netWorth } = data.netWorthInfo;
  return (
    <HorizontalContainer className="p-2">
      <LabelValue label="Net Worth" value={netWorth} />
      <LabelValue label="Investments" value={investments} />
      <LabelValue label="Cash" value={cash} />
    </HorizontalContainer>
  );
};
