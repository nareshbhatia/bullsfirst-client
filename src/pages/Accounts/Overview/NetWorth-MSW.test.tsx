import React from 'react';
import { render, waitForElementToBeRemoved } from '../../../test/test-utils';
import { NetWorth } from './NetWorth';

// mock react-router-dom
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    accountId: 'brokerage-account',
  }),
}));

describe('<NetWorth />', () => {
  test('renders correctly', async () => {
    const { asFragment, getByText } = render(<NetWorth />);

    await waitForElementToBeRemoved(getByText('Loading...'));

    expect(asFragment()).toMatchSnapshot();
  });
});
