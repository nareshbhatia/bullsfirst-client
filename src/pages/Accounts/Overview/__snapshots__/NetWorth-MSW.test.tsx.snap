// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NetWorth /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex-1 flex flex-row p-2"
    data-testid="horizontal-container"
  >
    <div
      class="net-worth__label-value"
    >
      <h1
        class="net-worth__label"
      >
        Net Worth
      </h1>
      <p
        class="net-worth__value"
      >
        $1,489,526.50
      </p>
    </div>
    <div
      class="net-worth__label-value"
    >
      <h1
        class="net-worth__label"
      >
        Investments
      </h1>
      <p
        class="net-worth__value"
      >
        $1,479,526.50
      </p>
    </div>
    <div
      class="net-worth__label-value"
    >
      <h1
        class="net-worth__label"
      >
        Cash
      </h1>
      <p
        class="net-worth__value"
      >
        $10,000.00
      </p>
    </div>
  </div>
</DocumentFragment>
`;
